gcc/rust/typecheck/rust-tyty.h

```diff
@@ -101,6 +101,70 @@ class BaseType
 
   virtual bool has_subsititions_defined () const { return false; }
 
+  virtual std::string to_string () const 
```

Comment:
I don't think we want a to_string implemented on TyTy::BaseType. This class already has a BaseType::to_string which seems like a conflict here.